### PR TITLE
Fix angry jenkins with broader exception handling

### PR DESCRIPTION
**Pull request by schottsfired**
* Auto-format `AWSCredentialsImpl.java`
* Change exception type from `AmazonServiceException` to `RuntimeException`

When garbage data is provided in the AWS credential configuration, the Amazon SDK throws a (previously uncaught) `SdkClientException` e.g. "unable to determine region ..." which caused an angry Jenkins to appear below the `secretKey` field. Now that the broad exception is caught, a much better looking error is shown which relays information about the `SdkClientException`.

Note: `AmazonServiceException` [is a subclass of](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/SdkClientException.html) `SdkClientException` and `RuntimeException`.

### CODE REVIEW

File: src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AWSCredentialsImpl.java [hunk 1]
```diff
@@ -136,9 +136,8 @@ public AWSCredentials getCredentials() {
             DefaultAwsRegionProviderChain sdkRegionLookup = new DefaultAwsRegionProviderChain();
             try {
                 clientRegion = sdkRegionLookup.getRegion();
-            }
-            catch(com.amazonaws.SdkClientException e) {
-                LOGGER.log(Level.WARNING,"Could not find default region using SDK lookup.", e);
+            } catch (com.amazonaws.SdkClientException e) {
```
Comment (jglick) [inline src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AWSCredentialsImpl.java:139]: or just

```suggestion
            } catch (RuntimeException e) {
```

?

File: src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AWSCredentialsImpl.java [hunk 2]
```diff
@@ -291,7 +290,7 @@ public FormValidation doCheckSecretKey(@QueryParameter("accessKey") final String
                             assumeResult.getCredentials().getAccessKeyId(),
                             assumeResult.getCredentials().getSecretAccessKey(),
                             assumeResult.getCredentials().getSessionToken());
-                } catch(AmazonServiceException e) {
+                } catch (SdkClientException e) {
```
Comment (jglick) [inline src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AWSCredentialsImpl.java:293]: (same)

File: src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AWSCredentialsImpl.java [hunk 3]
```diff
@@ -148,7 +147,7 @@ public AWSCredentials getCredentials() {
 
             AWSSecurityTokenService client;
             // Handle the case of delegation to instance profile
-            if (StringUtils.isBlank(accessKey) && StringUtils.isBlank(secretKey.getPlainText()) ) {
+            if (StringUtils.isBlank(accessKey) && StringUtils.isBlank(secretKey.getPlainText())) {
```
Comment (jglick) [inline src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AWSCredentialsImpl.java:150]: Not sure what the maintainer’s preferences are, but FWIW I advise keeping PRs as short as possible, meaning in particular do not make unrelated formatting changes.

File: src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AWSCredentialsImpl.java (hunk 1 above)
Comment (schottsfired) [reply src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AWSCredentialsImpl.java:139]: 🤷‍♂️ `SdkBaseException` works too: https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/SdkBaseException.html

Comment (jglick) [reply src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AWSCredentialsImpl.java:139]: Right. FWIW my general take on Java exceptions is: either your call worked or it failed, and if you are bothering to handle failures at some call site, just handle everything unless you have a specific reason to do otherwise.

File: src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AWSCredentialsImpl.java (hunk 2 above)
Comment (schottsfired) [reply src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AWSCredentialsImpl.java]: https://github.com/jenkinsci/aws-credentials-plugin/pull/76/commits/f5d29578dc343673b9f2454b42c3fbd62b6398bf

File: src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AWSCredentialsImpl.java (hunk 3 above)
Comment (schottsfired) [reply src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AWSCredentialsImpl.java:150]: Duly noted, I'll try not to do that in the future.

I'd un-format if I could, but I'm not sure how to do that.

(FWIW I'm using IntelliJ)

Comment (jglick) [reply src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AWSCredentialsImpl.java:150]: > I'd un-format if I could, but I'm not sure how to do that.

Well you can manually reverse-apply the portions of the diff which do not change behavior.

There is probably some setting in your IDE to refrain from changing formatting of a line unless you actually edited that line. I do not use IDEA so I cannot say for sure.

Comment (schottsfired) [reply src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AWSCredentialsImpl.java:150]: I'll mark this resolved for now. If someone (other than you) speaks up about the formatting, I'll manually change it back.
